lib/routes/account.js

```diff
@@ -18,16 +21,28 @@ module.exports = function (
   db,
   mailer,
   Password,
-  redirectDomain,
-  verifierVersion,
-  isProduction,
-  domain,
-  resendBlackoutPeriod,
+  config,
   customs,
   isPreVerified,
   checkPassword
   ) {
 
+  var openIdExtensions = [
+    new openid.AttributeExchange(
+      {
+        'http://axschema.org/contact/email': 'optional'
+      }
+    )
+  ]
+
+  function isOpenIdProviderAllowed(id) {
+    return config.openIdProviders.some(
+      function (allowed) {
+        return id.indexOf(allowed) === 0
```

Comment:
I see your point. This wasn't really meant as a security feature though. The url is user supplied in the UI for now. It was more like, "if you know one on our list, you can use it". But its not any more trouble to parse the url...
<img width="450" alt="screen shot 2015-08-19 at 7 30 04 pm" src="https://cloud.githubusercontent.com/assets/87619/9374240/bf6e0010-46a8-11e5-9600-d6f93797b9cd.png">